frontend/pages/pillars.vue

```diff
@@ -0,0 +1,198 @@
+﻿<script setup lang="ts">
+import { usePillars } from '~/composables/usePillars'
+import { ref } from 'vue'
+import type { GameDesign, Pillar } from '~/types/pillars'
+
+definePageMeta({
+  middleware: 'authentication',
+})
+
+const config = useRuntimeConfig()
+
+const {
+  pillars,
+  designIdea,
+  llmFeedback,
+  initalPillarFetch,
+  createPillar,
+  updatePillar,
+  deletePillar,
+  getLLMFeedback,
+  validatePillar,
+  updateDesignIdea,
+} = await usePillars()
+
+const selectedPillar = ref(-1)
+
+initalPillarFetch()
+
+await useFetch<GameDesign>(`${config.public.apiBase}/llm/design/get_or_create/`, {
+  method: 'GET',
+  credentials: 'include',
+  headers: useRequestHeaders(['cookie']),
+}).then((data) => {
+  designIdea.value = data.data.value?.description ?? ''
+})
+
+async function handlePillarCreation() {
+  const newPillar = await createPillar()
+  selectedPillar.value = newPillar.pillar_id
+}
+
+async function handlePillarSelect(toSelect: number) {
+  if (selectedPillar.value != -1) {
+    const pillar = pillars.value.find((p) => p.pillar_id === selectedPillar.value)
+    if (pillar) {
+      await updatePillar(pillar)
+    }
+  }
+  selectedPillar.value = toSelect
+}
+
+async function handlePillarDelete(pillar: Pillar) {
+  if (selectedPillar.value === pillar.pillar_id) {
+    selectedPillar.value = -1
+  }
+  await deletePillar(pillar)
+}
+</script>
+
+<template>
+  <div class="p-4">
+    <!-- Game Design Idea Section -->
+    <h2 class="text-2xl font-bold mb-4">Game Design Idea:</h2>
+    <div class="flex items-stretch w-300 max-w-xxl">
+      <UTextarea
+        v-model="designIdea"
+        placeholder="Enter your game design idea here..."
+        variant="outline"
+        color="secondary"
+        size="xl"
+        class="w-full"
+        autoresize
+        @focusout="updateDesignIdea"
+      />
+    </div>
+
+    <!-- Pillar Display -->
+    <h2 class="text-2xl font-bold mt-6 mb-4">Pillars:</h2>
+    <div class="flex mt-6 gap-4 flex-wrap p-5">
+      <div v-for="pillar in pillars" :key="pillar.pillar_id" class="relative">
+        <UCard class="w-70 min-h-55 flex flex-col justify-center-safe" variant="soft">
```

Comment:
not relevant for this merge request, but is probably worth to create a component for this